src/client/components/ClientState.tsx

```diff
@@ -1,9 +1,12 @@
 import React, { createContext, FunctionComponent } from 'react';
 import { ClientState } from '@/shared/model/ClientState';
 
-export const ClientStateContext = createContext({
+export const ClientStateContext = createContext<ClientState>({
   clientHosts: { idapiBaseUrl: '', oauthBaseUrl: '' },
   recaptchaConfig: { recaptchaSiteKey: '' },
+  queryParams: {
+    returnUrl: 'https://theguardian.com',
```

Comment:
can we set this based on the environment (code/prod)

Reply:
Unfortunately not, as this is on the client, it doesn't have access to the environment. However in most (if not all) cases this default value will not be reached as `returnUrl` is a required query parameter on every request.